### ISSUE TITLE
Postgres external table - wrong columns type

### ISSUE THREAD

**Issue by yoavo-datricks** [ClickHouse/ClickHouse#24804]
Hello.
When creating an external CH table to a PG table with an array data type the CH column is created as a string data type.

**How to reproduce**
Clickhouse docker 21.5.6.6-alpine
1. Create a PG table using the following create statement:
CREATE TABLE t1 (
	id varchar NULL,
	arr_t _int8 NULL
);
2. Create a CH table pointing the postgresql table:
CREATE or replace table "default"."ch_t1" as postgresql('pg_host', 'pg_db', 't1', 'pg_user', 'pg_password', 'pg_schema');
3. Execute in CH
describe "default"."ch_t1";

**Expected behavior**
The arr_t columns should be of type array and it is of type Nullable(String)

Thanks

**Comment by kssenii** [comment]
I am not sure there is a good solution for `arr_t_int8` data type.

Data types for a table are fetched with command `SELECT attname AS name, format_type(atttypid, atttypmod) AS type, attnotnull AS not_null, attndims AS dims  FROM pg_attribute WHERE attrelid = 't1'::regclass AND NOT attisdropped AND attnum > 0;  `.

For `attndims` always returns 0 dimensions for `arr_t_int8`, but it works fine for all other array data types. It is possible to get number of dimensions via `array_ndims`, but it cannot be properly joined with previous command - to fetch all info in one try. 

Without known number of dimensions it is not possible to construct clickhouse array data type, so it is converted to String as default for such cases.

**Comment by yoavo-datricks** [comment]
Hi. Thank you for your helpful response.
I tried testing the issue with other types and forms of "Create" statements and it seems PG has a bigger bug when it comes to attndims.

The following three create statements:

CREATE TABLE t1 (
  id varchar NULL,
  arr_t _int8 NULL
);

CREATE TABLE t2 (
  id varchar NULL,
  arr_t int8[] NULL
);

CREATE TABLE t3 as select * from t2;

Resulted in:
t1 arr_t attndims - 0
t2 arr_t attndims - 1
t3 arr_t attndims - 0

Tested it for int4, int8 and characters. All the same behavior.

What can we do? Is it possible to extend the code to check array_ndims on columns with square brackets since attndims is very unreliable?

Thanks!
